docs/nitpick-exceptions

```diff
@@ -72,58 +50,33 @@ py:class astropy.wcs.wcsapi.fitswcs.custom_ctype_to_ucd_mapping
 # numpy inherited docstrings
 py:obj dtype
 py:obj a
-py:obj a.size == 1
 py:obj n
 py:obj ndarray
 py:obj args
 
 # other classes and functions that cannot be linked to
-# NUMPY_LT_2_0: numpy 2.0 changed numpy.core to numpy._core
-py:class numpy.ma.core.MaskedArray
-py:class numpy.ma.mvoid
-py:class numpy.void
-py:class numpy.core.records.recarray
-py:class numpy._core.records.recarray
-py:class xmlrpclib.Fault
-py:class xmlrpclib.Error
-py:class xmlrpc.client.Fault
 py:class xmlrpc.client.Error
-py:obj pkg_resources.parse_version
-py:class pandas.DataFrame
 
 # Pending on python docs links issue #11975
-py:class list
-py:obj list.append
 py:obj list.append
 py:obj list.count
 py:obj list.extend
 py:obj list.index
 py:obj list.insert
 py:meth list.pop
 py:obj list.remove
-py:class classmethod
-py:obj RuntimeError
-py:obj NotImplementedError
-py:obj AttributeError
-py:obj NotImplementedError
 py:obj RendererBase
 py:obj Artist
 py:obj BboxBase
-py:obj Text
-py:obj text
 
 # This list is from https://github.com/numpy/numpydoc/issues/275
 py:class None.  Remove all items from D.
 py:class a set-like object providing a view on D's items
 py:class a set-like object providing a view on D's keys
-py:class v, remove specified key and return the corresponding value.
 py:class None.  Update D from dict/iterable E and F.
 py:class an object providing a view on D's values
 py:class a shallow copy of D
 
-# This extends the numpydoc list above to fix lincheck warning
```

Comment:
This comment mentions linkcheck, so I want to run linkcheck job but the skipping CI in commit has prevented this.